provider/default/app.js

```diff
@@ -1,50 +1,67 @@
-const map = new maplibregl.Map({
-  container: 'map',
-  hash: true,
-  style: `http://${window.location.host}/style.json`,
-})
-
-const socket = new WebSocket('ws://localhost:___PORT___')
-
-socket.addEventListener('message', (message) => {
-  map.setStyle(JSON.parse(message.data))
-})
-
-map.addControl(new maplibregl.NavigationControl(), 'top-right')
-
-map.addControl(
-  new MaplibreLegendControl(
-    {},
-    {
-      showDefault: true,
-      showCheckbox: true,
-      onlyRendered: true,
-      reverseOrder: true,
-    },
-  ),
-  'bottom-left',
-)
-
-const showTileBoundaries = document.getElementById('showTileBoundaries')
-const setShowTileBoundaries = function () {
-  const checked = showTileBoundaries.checked
-  map.showTileBoundaries = checked
-}
-setShowTileBoundaries()
-showTileBoundaries.addEventListener('click', setShowTileBoundaries)
-
-const showCollisionBoxes = document.getElementById('showCollisionBoxes')
-const setShowCollisionBoxes = function () {
-  const checked = showCollisionBoxes.checked
-  map.showCollisionBoxes = checked
-}
-setShowCollisionBoxes()
-showCollisionBoxes.addEventListener('click', setShowCollisionBoxes)
-
-const showPadding = document.getElementById('showPadding')
-const setShowPadding = function () {
-  const checked = showPadding.checked
-  map.showPadding = checked
-}
-setShowPadding()
-showPadding.addEventListener('click', setShowPadding)
+;(async () => {
+  const mapStyleUrl = `http://${window.location.host}/style.json`
+  const mapStyleRes = await fetch(mapStyleUrl)
+  const mapStyleJson = await mapStyleRes.json()
+  const mapTileUrl =
+    mapStyleJson['sources'][Object.keys(mapStyleJson['sources'])]['url']
+  const mapTileRes = await fetch(mapTileUrl)
+  const mapTileJson = await mapTileRes.json()
+  const bounds = mapTileJson.bounds
+  const center = mapTileJson.center
```

Comment:
I think it’s better to use bounds and center from style.json itself instead of calculating it from tilejson. Because sometimes users just use pbf source (not tilejson). I assume this code will not work except tilejson.

Reply:
Thank you all for your great efforts!
I think this is a good point. I do not often use tile json. 
And about the style.json, its "center" is optional, so we may need to keep the line 11 to 12 for style.json as well.